kratos/python_scripts/python_solver.py

```diff
@@ -0,0 +1,175 @@
+from __future__ import print_function, absolute_import, division  # makes KratosMultiphysics backward compatible with python 2.6 and 2.7
+
+# Importing Kratos
+import KratosMultiphysics
+
+# Other imports
+import os
+
+class PythonSolver(object):
+    """The base class for the Python Solvers in the applications
+    Changes to this BaseClass have to be discussed first!
+    """
+    def __init__(self, model, settings):
+        """The constructor of the PythonSolver-Object.
+
+        It is intended to be called from the constructor
+        of deriving classes:
+        super(DerivedSolver, self).__init__(settings)
+
+        Keyword arguments:
+        self -- It signifies an instance of a class.
+        model -- The Model to be used
+        settings -- The solver settings used
+        """
+        if (type(model) != KratosMultiphysics.Model):
+            raise Exception("Input is expected to be provided as a Kratos Model object")
+
+        if (type(settings) != KratosMultiphysics.Parameters):
+            raise Exception("Input is expected to be provided as a Kratos Parameters object")
+
+        self.model = model
+        self.settings = settings
+
+        self.echo_level = self.settings["echo_level"].GetInt()
+
+    def AddVariables(self):
+        """This function add the Variables needed by this PythonSolver to the the ModelPart
+        It has to be called BEFORE the ModelPart is read!
+        """
+        pass
+
+    def AddDofs(self):
+        """This function add the Dofs needed by this PythonSolver to the the ModelPart
+        It has to be called AFTER the ModelPart is read!
+        """
+        pass
+
+    def ReadModelPart(self):
+        """This function reads the ModelPart
+        """
+        raise Exception("This function has to be implemented in the derived class")
+
+    def PrepareModelPart(self):
+        """This function prepares the ModelPart for being used by the PythonSolver
+        """
+        pass
+
+    def GetMinimumBufferSize(self):
+        """This function returns the minimum buffer size needed for this PythonSolver
+        """
+        raise Exception('Please implement "GetMinimumBufferSize" in your derived solver')
+
+    def ImportModelPart(self):
```

Comment:
There are two of these, there is the ImportModelPart and the _ImportModelPart, why?

Reply:
OK, deprecated, I see